specification/keyvault/Security.KeyVault.Secrets/models.tsp

```diff
@@ -0,0 +1,433 @@
+   * retention interval (90 days), immediate and permanent deletion (i.e. purge) is
+   * not permitted, and in which the subscription itself  cannot be permanently
+   * canceled. System wil permanently delete it after 90 days, if not recovered
+   */
+  Recoverable_ProtectedSubscription: "Recoverable+ProtectedSubscription",
+
+  /**
+   * Denotes a vault state in which deletion is recoverable, and which also permits
+   * immediate and permanent deletion (i.e. purge when 7<= SoftDeleteRetentionInDays
+   * < 90). This level guarantees the recoverability of the deleted entity during
+   * the retention interval, unless a Purge operation is requested, or the
+   * subscription is cancelled.
+   */
+  CustomizedRecoverable_Purgeable: "CustomizedRecoverable+Purgeable",
+
+  /**
+   * Denotes a vault state in which deletion is recoverable without the possibility
+   * for immediate and permanent deletion (i.e. purge when 7<=
+   * SoftDeleteRetentionInDays < 90).This level guarantees the recoverability of the
+   * deleted entity during the retention interval and while the subscription is
+   * still available.
+   */
+  CustomizedRecoverable: "CustomizedRecoverable",
+
+  /**
+   * Denotes a vault and subscription state in which deletion is recoverable,
+   * immediate and permanent deletion (i.e. purge) is not permitted, and in which
+   * the subscription itself cannot be permanently canceled when 7<=
+   * SoftDeleteRetentionInDays < 90. This level guarantees the recoverability of the
+   * deleted entity during the retention interval, and also reflects the fact that
+   * the subscription itself cannot be cancelled.
+   */
+  CustomizedRecoverable_ProtectedSubscription: "CustomizedRecoverable+ProtectedSubscription",
+}
+
+/**
+ * The secret set parameters.
+ */
+model SecretSetParameters {
+  /**
+   * The value of the secret.
+   */
+  value: string;
+
+  /**
+   * Application specific metadata in the form of key-value pairs.
+   */
+  tags?: Record<string>;
+
+  /**
+   * Type of the secret value such as a password.
+   */
+  contentType?: string;
+
+  /**
+   * The secret management attributes.
+   */
+  @encodedName("application/json", "attributes")
+  secretAttributes?: SecretAttributes;
+}
+
+/**
+ * The secret management attributes.
+ */
+model SecretAttributes {
+  ...Attributes;
+
+  /**
+   * softDelete data retention days. Value should be >=7 and <=90 when softDelete
+   * enabled, otherwise 0.
+   */
+  @visibility("read")
+  recoverableDays?: int32;
+
+  /**
+   * Reflects the deletion recovery level currently in effect for secrets in the
+   * current vault. If it contains 'Purgeable', the secret can be permanently
+   * deleted by a privileged user; otherwise, only the system can purge the secret,
+   * at the end of the retention interval.
+   */
+  @visibility("read")
+  recoveryLevel?: string; // DeletionRecoveryLevel, but this should be treated as an opaque string.
```

Comment:
👋 I think this should be `DeletionRecoveryLevel` and not a `string` IIUC - because `DeletionRecoveryLevel` is not actually used in routes it is not generated (at least by the typespec-ts codegen) - can we change that? 

When I change it to DeletionRecoveryLevel I see the correct code being generated (including extensible enum patterns)

Reply:
We should be able to get `DeletionRecoveryLevel` generated by using `@@access(DeletionRecoveryLevel, Access.public)` in our `client.tsp`. For example, from another library: https://github.com/Azure/azure-rest-api-specs/blob/cf80ee4aacdb4ef9926b42f09837fe16add9e66a/specification/ai/Face/client.tsp#L29

Reply:
But why not use DeletionRecoveryLevel here instead of adding a client.tsp customization - does it not generate the correct thing? Just trying to understand the reasoning behind not using `recoveryLevel?: DeletionRecoveryLevel`

Reply:
Using a plain string instead of the enum was a [change request](https://teams.microsoft.com/l/message/19:eed5a67bbf6649238d7f802f324aa2ea@thread.skype/1728944022005?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1728944022005&teamName=Azure%20SDK&channelName=Service%20-%20KeyVault&createdTime=1728944022005) from Heath; "[i]t's response-only and should not be compared, which is why none of our client libraries should ship it as an enum".

Reply:
Ah ok, got it! Thanks for the context. 

So @heaths do you consider it a bug that JS shipped `KnownDeletionRecoveryLevel` as part of our [public API](https://github.com/Azure/azure-sdk-for-js/blob/ae05a39d0f9e313e82fc93ed37fa5c3c09e17db0/sdk/keyvault/keyvault-secrets/review/keyvault-secrets.api.md?plain=1#L70-L78)? I think what you said makes sense re: response-only just want to triple-check - if an extensible enum value can only show up in a response, never as an input, we should not be shipping the `Known` enum values as part of our public API is that accurate?

Reply:
To close the loop on this, we reached out to the service team and were told directly that the `DeletionRecoveryLevel` values are stable and can thus be used as the type for this attribute. I reverted the change to match the original spec across all libraries.